airsonic-main/src/main/java/org/airsonic/player/service/AudioScrobblerService.java

```diff
@@ -18,309 +18,60 @@
  Based upon Subsonic, Copyright 2009 (C) Sindre Mehus
  */
 package org.airsonic.player.service;
-
 import org.airsonic.player.domain.MediaFile;
 import org.airsonic.player.domain.UserSettings;
-import org.airsonic.player.util.StringUtil;
-import org.apache.commons.codec.digest.DigestUtils;
-import org.apache.http.NameValuePair;
-import org.apache.http.client.ResponseHandler;
-import org.apache.http.client.config.RequestConfig;
-import org.apache.http.client.entity.UrlEncodedFormEntity;
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.client.methods.HttpUriRequest;
-import org.apache.http.impl.client.BasicResponseHandler;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClients;
-import org.apache.http.message.BasicNameValuePair;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.airsonic.player.service.scrobbler.LastFMScrobbler;
+import org.airsonic.player.service.scrobbler.ListenBrainzScrobbler;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import java.io.IOException;
-import java.net.URI;
-import java.util.*;
-import java.util.concurrent.LinkedBlockingQueue;
+import java.util.Date;
 
 /**
  * Provides services for "audioscrobbling", which is the process of
- * registering what songs are played at www.last.fm.
- * <p/>
- * See http://www.last.fm/api/submissions
+ * registering what songs are played at website.
  *
- * @author Sindre Mehus
+ * @author Shen-Ta Hsieh
```

Comment:
Seeing this I'm not sure about adding names to the source code. Down the road a file may be modified by a lot of people, so it's definitely not going to be one person ; plus it's available from the Git history.